frame/fee-market/src/s2s/payment.rs

```diff
@@ -123,88 +130,89 @@ where
 	let mut assigned_relayers_rewards = BTreeMap::<T::AccountId, RingBalance<T>>::new();
 	let mut treasury_total_rewards = RingBalance::<T>::zero();
 
-	for (lane_id, message_nonce) in <ConfirmedMessagesThisBlock<T>>::get() {
-		// The order created when message was accepted, so we can always get the order info below.
-		if let Some(order) = <Orders<T>>::get(&(lane_id, message_nonce)) {
-			// The confirm_time of the order is set in the `OnDeliveryConfirmed` callback. And the callback function
-			// was called as source chain received message delivery proof, before the reward payment.
-			let order_confirm_time = order
-				.confirm_time
-				.unwrap_or_else(|| frame_system::Pallet::<T>::block_number());
-			// Iterate the unrewarded relayer list to get message relayer
-			let mut message_relayer = T::AccountId::default();
-			for unrewarded_relayer in messages_relayers.iter() {
-				if unrewarded_relayer.messages.contains_message(message_nonce) {
-					message_relayer = unrewarded_relayer.relayer.clone();
-					break;
-				}
-			}
+	for entry in messages_relayers {
+		let nonce_begin = sp_std::cmp::max(entry.messages.begin, *received_range.start());
+		let nonce_end = sp_std::cmp::min(entry.messages.end, *received_range.end());
 
-			match order.lowest_and_highest_fee() {
-				(Some(lowest_fee), Some(message_fee)) => {
-					let message_reward;
-					let confirm_reward;
+		for message_nonce in nonce_begin..nonce_end + 1 {
+			// The order created when message was accepted, so we can always get the order info below.
+			if let Some(order) = <Orders<T>>::get(&(lane_id, message_nonce)) {
+				// The confirm_time of the order is set in the `OnDeliveryConfirmed` callback. And the callback function
+				// was called as source chain received message delivery proof, before the reward payment.
+				let order_confirm_time = order
```

Comment:
@AsceticBear Here, order's confirm_time is supposed to be modified with order_confirm_time right? But I can not find the modification code.

Reply:
We modified the confirmed time in the callback. See https://github.com/darwinia-network/darwinia-common/blob/change-code-to-adapt-upstream/frame/fee-market/src/s2s/callbacks.rs#L60